### ISSUE TITLE
Already logged in user users can visit accounts/signin

### ISSUE THREAD

**Issue by sinergatis** [aristippe/pathagar#42]
Currently, if a user is already logged in, he is allowed to visit the login url at `/accounts/signin/`, and the results are a bit strange: the username/password box is displayed, along with the standard header and footer for logged in users.

It might be by [userena design](https://github.com/bread-and-pepper/django-userena/issues/95), but perhaps some setting can be toggled somewhere on the later versions of userena to prevent this behavior? Other alternatives would be to tweak the template so the form is not displayed for logged in users, or wrap the `signin` view onto another one that basically redirects the user to `home` or similar.

**Comment by aristippe** [comment]
Good find. It is indeed because of userena, though the same thing might have happened with the older login place. Apparently a server redirect is [recommended](https://www.w3.org/QA/Tips/reback). Maybe it's possible to override the default `userena.views.signin` (or report a bug), or go back to the old sign in view and add a redirect there.

**Comment by sinergatis** [comment]
Yeah, mainly a matter of digging through userena docs and issues.

If nothing is found, I'd go indeed with writing a small login view that basically checks if the user is logged in: if it is, just redirect him to home; if not, redirected him (or call) userena.views.signin.

**Comment by aristippe** [comment]
I looked at the signin function of userena and it doesn't do anything if already logged in. I filed an [issue](https://github.com/bread-and-pepper/django-userena/issues/516). I'll also lookup the old login code and give it a try.

**Comment by aristippe** [comment]
Haven't had a chance to look at it recent though someone did comment the issue:

https://github.com/bread-and-pepper/django-userena/issues/516#issuecomment-201304151
